test/linalg/diagonal.jl

```diff
@@ -312,7 +312,7 @@ end
 end
 
 # allow construct from range
-@test Diagonal(linspace(1,3,3)) == Diagonal([1.,2.,3.])
+@test Diagonal(linspace(1,3,3)) ≈ Diagonal([1.,2.,3.])
```

Comment:
These should be elementwise equal - is this better expressed as `all(Diagonal(linspace(1,3,3)) .== Diagonal([1.0,2.0,3.0]))`?

I must admit to being surprised that `1:2 == [1,2]` is false in the first place, as other `AbstractVector`s are compared elementwise.  I hunted this down to https://github.com/JuliaLang/julia/issues/5778 which seems to be a bit of a thorny issue.